source/common/http/filter_manager.cc

```diff
@@ -1174,6 +1175,12 @@ void FilterManager::encodeData(ActiveStreamEncoderFilter* filter, Buffer::Instan
       trailers_added_entry = entry;
     }
 
+    if (state_.local_reply_stream_reset_) {
+      // If the stream is reset by a call to sendLocalReply, we ship out the reply directly.
```

Comment:
Thanks! This makes much more sense and is generalized. Switched it to use `state_.destroyed_`.

Reply:
I had to move the short-circuit logic back into `commonHandleAfterDataCallback`. Short-circuiting before with the `state_.destroyed` broke a HTTP conn manager test for buffer limits, because it has to handle the `StopIterationAndBuffer` case.

https://github.com/envoyproxy/envoy/blob/e20957c011b11a4f9f935ec7556ed89d1d8c22d7/test/common/http/conn_manager_impl_test_2.cc#L1403

```

test/common/http/conn_manager_impl_test_2.cc:1433
Failed
Logs:
 [2021-03-11 18:49:07.433][14][trace][http] [source/common/http/filter_manager.cc:1190] [C71][S0] encode data called: filter=0xdb93f6ab810 status=1
 Do NOT contain:
  'debug', 'Resetting stream due to response_payload_too_large. Prior headers have already been sent'
Stack trace:
  0x1be5a18: Envoy::Http::HttpConnectionManagerImplTest_HitResponseBufferLimitsAfterHeaders_Test::TestBody()
  0x417abe4: testing::internal::HandleSehExceptionsInMethodIfSupported<>()
  0x416acbe: testing::internal::HandleExceptionsInMethodIfSupported<>()
  0x41530b3: testing::Test::Run()
  0x4153cad: testing::TestInfo::Run()
... Google Test internal frames ...

test/common/http/conn_manager_impl_test_2.cc:1428
Actual function call count doesn't match EXPECT_CALL(stream_, resetStream(_))...
         Expected: to be called once
           Actual: never called - unsatisfied and active
Stack trace:
  0x4127fe9: testing::internal::GoogleTestFailureReporter::ReportFailure()
  0x1b521a8: testing::internal::Expect()
  0x412db6e: testing::internal::UntypedFunctionMockerBase::VerifyAndClearExpectationsLocked()
  0x22b3100: testing::internal::FunctionMocker<>::~FunctionMocker()
  0x22b0cd3: Envoy::Http::MockStream::~MockStream()
  0x1ca3d81: Envoy::Http::HttpConnectionManagerImplTest::~HttpConnectionManagerImplTest()
  0x1c53f88: Envoy::Http::HttpConnectionManagerImplTest_HitResponseBufferLimitsAfterHeaders_Test::~HttpConnectionManagerImplTest_HitResponseBufferLimitsAfterHeaders_Test()
  0x1c53fac: Envoy::Http::HttpConnectionManagerImplTest_HitResponseBufferLimitsAfterHeaders_Test::~HttpConnectionManagerImplTest_HitResponseBufferLimitsAfterHeaders_Test()
  0x416b69b: testing::Test::DeleteSelf_()
  0x417abe4: testing::internal::HandleSehExceptionsInMethodIfSupported<>()
  0x416acbe: testing::internal::HandleExceptionsInMethodIfSupported<>()
  0x4153cf8: testing::TestInfo::Run()
  0x415457c: testing::TestSuite::Run()
... Google Test internal frames ...
```